commonmark/src/main/java/org/commonmark/internal/inline/HtmlInlineParser.java

```diff
@@ -0,0 +1,210 @@
+                    if (scanner.find('\'') < 0) {
+                        return false;
+                    }
+                    scanner.next();
+                } else if (valueStart == '"') {
+                    scanner.next();
+                    if (scanner.find('"') < 0) {
+                        return false;
+                    }
+                    scanner.next();
+                } else {
+                    if (scanner.find(attributeValueEnd) <= 0) {
+                        return false;
+                    }
+                }
+
+                // Whitespace is required between attributes
+                whitespace = scanner.whitespace() >= 1;
+            }
+        }
+
+        scanner.next('/');
+        return scanner.next('>');
+    }
+
+    private static boolean tryClosingTag(Scanner scanner) {
+        // spec: A closing tag consists of the string </, a tag name, optional whitespace, and the character >.
+        scanner.next();
+        if (scanner.match(tagNameStart) >= 1) {
+            scanner.match(tagNameContinue);
+            scanner.whitespace();
+            return scanner.next('>');
+        }
+        return false;
+    }
+
+    private static boolean tryProcessingInstruction(Scanner scanner) {
+        // spec: A processing instruction consists of the string <?, a string of characters not including the string ?>,
+        // and the string ?>.
+        scanner.next();
+        while (scanner.find('?') > 0) {
+            scanner.next();
+            if (scanner.next('>')) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean tryComment(Scanner scanner) {
+        // spec: An HTML comment consists of <!-- + text + -->, where text does not start with > or ->, does not end
+        // with -, and does not contain --. (See the HTML5 spec.)
+
+        // Skip first `-`
+        scanner.next();
+        if (!scanner.next('-')) {
+            return false;
+        }
+
+        if (scanner.next('>')) {
+            return false;
+        }
+
+        if (scanner.next('-')) {
+            // Can't start with ->
+            if (scanner.next('>')) {
+                return false;
+            }
+            // Empty comment
+            if (scanner.next('-')) {
+                return scanner.next('>');
+            }
+        }
+
+        while (scanner.find('-') >= 0) {
+            if (scanner.next('-') && scanner.next('-')) {
+                return scanner.next('>');
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean tryCdata(Scanner scanner) {
+        // spec: A CDATA section consists of the string <![CDATA[, a string of characters not including the string ]]>,
+        // and the string ]]>.
+
+        // Skip `[`
+        scanner.next();
+
+        if (scanner.next('C') && scanner.next('D') && scanner.next('A') && scanner.next('T') && scanner.next('A')
+                && scanner.next('[')) {
```

Comment:
To have an overload `next(...)` method for multiple char could increase even more the readability here.

Something like
```java
    if (scanner.next('C', 'D', 'A', 'T', 'A', '[') {
   ....

    //Scanner.java
    public boolean next(char... c) {
        for (char character: c) {
            if (!isNext(character)){
                return false;
            }
        }
        return true;
    }

   //ScannerTest.java
    @Test
    public void testIsNextMultipleCharacter() {
        Scanner scanner = new Scanner(Collections.<CharSequence>singletonList("foo bar"), 0, 0);
        assertTrue(scanner.isNext('f', 'o', 'o', ' ', 'b', 'a', 'r'));
        assertFalse(scanner.isNext('r'));
    }
```

Reply:
Yeah, I was considering this before, maybe even a `next(String)` instead, which only advances if it finds the whole string.